parsers/CA_PE.py

```diff
@@ -1,160 +1,147 @@
+
+    event = _get_event(session or Session())
+
+    production_mix = ProductionMix()
+    # Specify some sources that definitely aren't present on PEI as zero; this
+    # allows the analyzer to better estimate CO2eq.
+    production_mix.add_value("coal", 0)
+    production_mix.add_value("geothermal", 0)
+    production_mix.add_value("hydro", 0)
+    production_mix.add_value("nuclear", 0)
+    # These are oil-fueled ("heavy fuel oil" and "diesel") generators used as
+    # peakers and back-up.
+    production_mix.add_value("oil", event["fossil"])
+    production_mix.add_value("wind", event["wind"])
+
+    production_breakdowns = ProductionBreakdownList(logger)
+    production_breakdowns.append(
+        datetime=event["datetime"],
+        production=production_mix,
+        source=SOURCE,
+        zoneKey=ZONE_KEY,
+    )
+    return production_breakdowns.to_list()
 
 
 def fetch_exchange(
-    zone_key1: str,
-    zone_key2: str,
+    zone_key1: ZoneKey,
+    zone_key2: ZoneKey,
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-) -> dict[str, Any]:
+) -> list[dict[str, Any]]:
     """Requests the last known power exchange (in MW) between two regions."""
     if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
-
-    sorted_zone_keys = "->".join(sorted([zone_key1, zone_key2]))
-
-    if sorted_zone_keys != "CA-NB->CA-PE":
-        raise NotImplementedError("This exchange pair is not implemented")
-
-    requests_obj = session or Session()
-    pei_info = _get_pei_info(requests_obj)
-
-    if pei_info is None or pei_info["pei_load"] is None:
-        return None
-
-    # PEI imports most of its electricity. Everything not generated on island
-    # is imported from New Brunswick.
-    # In case of wind, some is paper-"exported" even if there is a net import,
-    # and 'pei_wind_used'/'data5' indicates their accounting of part of the load
-    # served by non-exported wind.
-    # https://www.princeedwardisland.ca/en/feature/pei-wind-energy says:
-    # "Wind Power Exported Off-Island is that portion of wind generation that is supplying
-    # contracts elsewhere. The actual electricity from this portion of wind generation
-    # may stay within PEI but is satisfying a contractual arrangement in another jurisdiction."
-    # We are ignoring these paper exports, as they are an accounting/legal detail
-    # that doesn't actually reflect what happens on the wires.
-    # (New Brunswick being the only interconnection with PEI, "exporting" wind power to NB
-    # then "importing" a balance of NB electricity likely doesn't actually happen.)
-    imported_from_nb = (
-        pei_info["pei_load"] - pei_info["pei_fossil_gen"] - pei_info["pei_wind_gen"]
+        raise ParserException("CA_PE.py", "Unable to fetch historical data", ZONE_KEY)
```

Comment:
The zone_key here should be the exchange key.

Reply:
Uh oh, it looks like this didn't get corrected in time. Do we need another PR?

Reply:
Ah I missed my own comment! 😅

It's probably fine but if you wanted you could add a consumption parser that takes the load data and returns that (and slip in this change).

I added it to my todo list but if you want to do it feel free to do so!

Reply:
I see you've fixed this one too, thanks again!